python/ray/tune/execution/experiment_state.py

```diff
@@ -122,22 +122,34 @@ class _ExperimentCheckpointManager:
     def __init__(
         self,
         *,
-        local_checkpoint_dir: str,
-        remote_checkpoint_dir: str,
         checkpoint_period: Union[int, float, str],
-        sync_config: SyncConfig,
+        # TODO(justinvyu): Make `storage` required
+        storage: Optional[StorageContext] = None,
         sync_every_n_trial_checkpoints: Optional[int] = None,
+        # TODO(justinvyu): Remove these args.
+        sync_config: SyncConfig = None,
+        local_checkpoint_dir: str = None,
+        remote_checkpoint_dir: str = None,
     ):
-        # Checkpoint directories
-        self._local_checkpoint_dir = local_checkpoint_dir
-        self._remote_checkpoint_dir = remote_checkpoint_dir
-
-        # Synch to/from cloud
-        self._sync_config = sync_config or SyncConfig
-        # Resolves syncer="auto" to an actual syncer if needed
-        self._syncer = get_node_to_storage_syncer(
-            self._sync_config, self._remote_checkpoint_dir
-        )
+        self._storage = storage
+        if _use_storage_context():
+            assert storage
+
+            self._local_checkpoint_dir = None
+            self._remote_checkpoint_dir = None
+            self._sync_config = None
+            self._syncer = None
+        else:
+            # Checkpoint directories
+            self._local_checkpoint_dir = local_checkpoint_dir
+            self._remote_checkpoint_dir = remote_checkpoint_dir
```

Comment:
```suggestion
            self._legacy_remote_checkpoint_dir = remote_checkpoint_dir
```

It might help to prepend these attributes with legacy_ as we refactor classes to keep old vs new paths straight.